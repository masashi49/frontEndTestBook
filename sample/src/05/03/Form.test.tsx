import { fireEvent,logRoles,render,screen } from '@testing-library/react';
import { Form } from './Form';

test('名前の表示', () => {
  render(<Form name='taro' />);
  expect(screen.getByText('taro')).toBeInTheDocument();
});

test('ボタンの表示', () => {
  render(<Form name='taro' />);
  expect(screen.getByRole('button')).toBeInTheDocument();
});

test('見出しの表示', () => {
  render(<Form name='taro' />);
  expect(screen.getByRole('heading')).toHaveTextContent('アカウント情報');
});

test('ボタンを押下すると、イベントハンドラーが呼ばれる', () => {
  const mockFn = jest.fn();
  render(<Form name='taro' onSubmit={mockFn} />);
  fireEvent.click(screen.getByRole('button'));
  expect(mockFn).toHaveBeenCalled();
});

test('Snapshot: taitoru 「リスト」が表示される', () => {
  const { container } = render(<Form title='リスト' />);
  expect(container).toMatchSnapshot();
});

test('logRoles: アクセシブルネームを確認する', () => {
  const { container } = render(<Form title='リスト' />);
  logRoles(container);
});
