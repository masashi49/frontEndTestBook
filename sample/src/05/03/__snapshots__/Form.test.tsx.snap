// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: taitoru 「リスト」が表示される 1`] = `
<div>
  <form>
    <h2>
      タイトル
    </h2>
    <p>
      リスト
    </p>
    <div>
      <button>
        決定
      </button>
    </div>
  </form>
</div>
`;
